eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventData.java

```diff
@@ -344,7 +412,7 @@ public long sequenceNumber() {
          * @return The name of the publisher. Or {@code null} if this was not sent to a publisher endpoint.
          */
         public String publisher() {
-            return this.getSystemProperty(PUBLISHER_ANNOTATION_NAME);
+            return this.getSystemProperty(PUBLISHER_ANNOTATION_NAME.getValue());
```

Comment:
You plan to use the Configuration API that Alan built recently, rather than going straight to getting system properties?

Reply:
These system properties are deceiving that they are properties on an AMQP message that are returned by the Event Hubs service. It's not related to the environment/configuration system properties.

Reply:
Ah yes, fooled again! Resolved! :-)